### ISSUE TITLE
Restrict Special Characters/Symbols in User Names (& Other Fields?)

### ISSUE THREAD

**Issue by SimoneZaza** [alkem-io/client-web#6872]
## Description
As a security-conscious product owner, I want to prevent users from using special characters/symbols in their usernames and other fields, So that we can mitigate the risk of malicious scripts being executed through user inputs.

## Acceptance criteria

- [ ] Should restrict the use of special characters/symbols in usernames.
- [ ] TBD Should restrict the use of special characters/symbols in other specified fields?
- [ ] Should allow only the following special characters in usernames and other fields:
- Periods: .
- Underscores: _
- Dashes: -
- [ ] Should display a user-friendly error message when a user attempts to use restricted characters.
- Error: Invalid Characters. Please use only letters, numbers, periods (.), underscores (_), and dashes (-).
- [ ] Should ensure that existing usernames and fields are sanitized to remove any restricted special characters/symbols.

## Additional Context
At least the following symbols should be restricted:

- Ampersand: &
- Equals sign: =
- Apostrophe: ’
- Plus sign: +
- Comma: ,
- Brackets: <, >

## Areas that will be affected 
 
_To be added during the refinement_

**Comment by techsmyth** [comment]
@SimoneZaza also important to have checks server side as well to prevent abuse via api. Client logic is good user experience but not actually secure.

**Comment by hero101** [comment]
@hoyterutteman i don't believe this has the correct parent since it's not a bug OR squashable in any means. I think this has to go through the normal planning process since it's requires to effort and considerations.

**Comment by hoyterutteman** [comment]
Agree, will take it out
